execution/function/scalar.go

```diff
@@ -6,15 +6,32 @@ package function
 import (
 	"context"
 	"math"
+	"time"
 
 	"github.com/prometheus/prometheus/model/labels"
 
 	"github.com/thanos-io/promql-engine/execution/model"
 )
 
 type scalarFunctionOperator struct {
-	pool *model.VectorPool
-	next model.VectorOperator
+	pool     *model.VectorPool
+	next     model.VectorOperator
+	ti       model.TimingInformation
+	duration time.Duration
+}
+
+func (o *scalarFunctionOperator) Analyze() (*model.TimingInformation, []model.ObservableVectorOperator) {
```

Comment:
Hmm, we should probably hook this up to something like the `ExplainableQuery` interface.